src/ConfigManager.cpp

```diff
@@ -235,7 +237,7 @@ void ConfigManager::clearWifiSettings(bool reboot) {
   memset(password, 0, PASSWORD_LENGTH);
 
   DebugPrintln(F("Clearing WiFi connection."));
-  storeWifiSettings(ssid, password, true);
+  storeWifiSettings(ssid, password, false);
```

Comment:
When clearing wifi settings, the magicbyte was being set to the emptyByte. 

I feel this had a reason a few versions ago. But currently when doing this it would no longer read the managed config. It seems like there is no reason for the magicByte to be set to the `magicBytesEmpty` .. Currently checking for the magicByte identifies if the memory has been initialized, so resetting it doesnt make sense.

In order to not try to connect to a non-configued SSID ("") I added the guard around the connection logic.

The new flow is therefore
# PR
- Read EEPROM
- Check that magicByte is there
- If not, jumps over to APmode
- If its there, reads SSID and configs
- If not SSID, jumps to APmode (now with configs available) 
- If SSID -- attempt to connect to WeeFee

# Current
- Read EEPROM
- Check that magicByte is there
- If not, jumps over to APmode
- If its there, reads SSID and configs ** This is not great if purposefully wanting to be in APmode with configs
- If SSID -- attempt to connect to WeeFee

# Alternative

- Move the readConfigs() outside the magicByte check

I find this odd, as then you dont know if the memory was initialized or this is first run .. This is hard to explain, hopefully the changes make more sense than my babble

Reply:
ooooh .. this just gave me an idea .. you once wrote [The system has no concept of "first load"](https://github.com/nrwiersma/ConfigManager/issues/46#issuecomment-552740459) ... Using the magicByte as initialization indicator like this would allow something like .. which I could work on ... 

```
void setup() {
  configManager.setInitCallback(myInitStuff);
}

void myInitStuff() {
  I dont know, maybe do some defaults ???
}


------- configManager.begin()

if (memcmp(magic, magicBytes, MAGIC_LENGTH) == 0) {

...
current connection logic 

} else if (initCallback) {
    initCallback();
}
```

It appears there is already a comment thinking the same 

```
// We are at a cold start, don't bother timing out.
```